packages/vite/src/node/plugins/worker.ts

```diff
@@ -8,6 +8,38 @@ import path from 'path'
 import { onRollupWarning } from '../build'
 
 const WorkerFileId = 'worker_file'
+export const inlineWorkerLoaderId = `vite/inline-worker-loader`
+
+// run in client load inline worker
+export function inlineWorkerLoader(
+  workerConstructor: FunctionConstructor,
+  workerOptions: any,
+  encodedJs: string
+) {
+  const blob =
+    // @ts-ignore
+    typeof window !== 'undefined' &&
+    // @ts-ignore
+    window.Blob &&
+    // @ts-ignore
+    new Blob([atob(encodedJs)], { type: 'text/javascript;charset=utf-8' })
+  // @ts-ignore
+  const objURL = blob && (window.URL || window.webkitURL).createObjectURL(blob)
+  try {
+    return objURL
+      ? new workerConstructor(objURL, workerOptions)
+      : new workerConstructor(
+          'data:application/javascript;base64,' + encodedJs,
+          workerOptions
+        )
+  } finally {
+    // revokeObjectURL in nextTick
+    setTimeout(() => {
+      // @ts-ignore
+      objURL && (window.URL || window.webkitURL).revokeObjectURL(objURL)
+    })
```

Comment:
there also can't exec `revokeObjectURL` in the right time.

Reply:
I'm not sure if I can provide an `?worker-bundle` to execute the bundle script, allowing users to create inline workers themselves.